src/DotNetInternals/DotNetInternals.csproj

```diff
@@ -1,23 +1,38 @@
-<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
-
-  <PropertyGroup>
-    <TargetFramework>net9.0</TargetFramework>
-    <Nullable>enable</Nullable>
-    <ImplicitUsings>enable</ImplicitUsings>
-    <ServiceWorkerAssetsManifest>service-worker-assets.js</ServiceWorkerAssetsManifest>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" />
-    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" PrivateAssets="all" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <ServiceWorker Include="wwwroot\service-worker.js" PublishedContent="wwwroot\service-worker.published.js" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <ProjectReference Include="..\RazorAccess\RazorAccess.csproj" />
-  </ItemGroup>
+﻿<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+
+	<PropertyGroup>
+		<TargetFramework>net9.0</TargetFramework>
+		<Nullable>enable</Nullable>
+		<ImplicitUsings>enable</ImplicitUsings>
+		<ServiceWorkerAssetsManifest>service-worker-assets.js</ServiceWorkerAssetsManifest>
+	</PropertyGroup>
+
+	<ItemGroup>
+	  <None Remove="Npm\src\vim-mode.js" />
+	</ItemGroup>
+
+	<ItemGroup>
+	  <Content Include="Npm\src\vim-mode.js" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<PackageReference Include="Blazored.LocalStorage" />
+		<PackageReference Include="BlazorMonaco" />
+		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" />
+		<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" PrivateAssets="all" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<ServiceWorker Include="wwwroot\service-worker.js" PublishedContent="wwwroot\service-worker.published.js" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<ProjectReference Include="..\RazorAccess\RazorAccess.csproj" />
+	</ItemGroup>
+
+	<Target Name="PreBuild" BeforeTargets="PreBuildEvent">
+		<Exec Command="npm install" WorkingDirectory="Npm" />
+		<Exec Command="npm run build" WorkingDirectory="Npm" />
```

Comment:
All that we _should_ need to do is install NPM before running `dotnet build`. I used the latest node, 22.X

Reply:
> I assume there's also some infra work that will need to be done to enable the node module to be built; node/npm needs to be installed so that it can be run as part of the build process, but I'm not sure how the build infra here is actually working.

It's built on Cloudflare Pages and Node.js is already there: https://developers.cloudflare.com/pages/configuration/language-support-and-tools/. We could just specify version if needed (in .nvmrc or .node-version file).